File: dotnet/Gherkin/Token.cs
using Gherkin.Ast;

namespace Gherkin;

public class Token
{
    public Token(GherkinLine line, Location location)
    {
        Line = line;
        Location = location;
    }
    public Token(Location location)
    {
        Location = location;
        IsEOF = true;
    }

    public bool IsEOF { get; }
    public GherkinLine Line { get; set; }
    public TokenType MatchedType { get; set; }
    public string MatchedKeyword { get; set; }
    public string MatchedText { get; set; }
    public int MatchedIndent { get; set; }
    public GherkinDialect MatchedGherkinDialect { get; set; }
    public Location Location { get; set; }

    public void Detach()
    {
    }

    public string GetTokenValue()
    {
        return IsEOF ? "EOF" : Line.GetLineText(-1);
    }

    public override string ToString()
    {
        return string.Format("{0}: {1}/{2}", MatchedType, MatchedKeyword, MatchedText);
    }
}

File: python/gherkin/token.py
from .gherkin_line import GherkinLine
from .parser_types import Location


class Token:
    def __init__(self, gherkin_line: GherkinLine, location: Location):
        self.line = gherkin_line
        self.location = location

    def eof(self) -> bool:
        return not self.line

    def detach(self) -> None:
        pass  # TODO: detach line - is this needed?

    def token_value(self) -> str:
        return "EOF" if self.eof() else self.line.get_line_text()
